src/java.base/share/classes/java/util/Range.java

```diff
@@ -0,0 +1,1008 @@
+package java.util;
+
+/**
+ * Represents a range of values.
+ *
+ * @param <T> the type of the values
+ */
+public abstract sealed class Range<T extends Comparable<T>> {
+    /**
+     * Returns an unbounded range.
+     *
+     * @param <T> the type of the values
+     * @return an unbounded range
+     */
+    public static <T extends Comparable<T>> Range<T> unbounded() {
+        return new UnboundedRange<>();
+    }
+
+    /**
+     * Returns an unbounded range ending at the given value.
+     *
+     * @param end the end of the range
+     * @param <T> the type of the values
+     * @return an unbounded range ending at the given value
+     */
+    public static <T extends Comparable<T>> Range<T> unboundedEndingAt(T end) {
+        return new UnboundedStartRange<>(end);
+    }
+
+    /**
+     * Returns an unbounded range ending at the given value.
+     *
+     * @param end the end of the range
+     * @param excludeEnd whether the end is exclusive
+     * @param <T> the type of the values
+     * @return an unbounded range ending at the given value
+     */
+    public static <T extends Comparable<T>> Range<T> unboundedEndingAt(T end, boolean excludeEnd) {
+        return excludeEnd ? new UnboundedStartRange<>(end, true) : new UnboundedStartRange<>(end);
```

Comment:
```suggestion
        return new UnboundedStartRange<>(end, excludeEnd);
```

Reply:
Really, didnt even notice. kind of ashamed of that obvious mistake honestly. Thank you